js/catalogo.js

```diff
@@ -0,0 +1,137 @@
+export class Producto {
+    constructor(id, nombre, precio, cantidad, categoria, uso, marca, tipo) {
+        this.id = id;
+        this.nombre = nombre;
+        this.precio = precio;
+        this.cantidad = cantidad;
+        this.categoria = categoria; // piel: seca, mixta, grasa, normal, sensible
+        this.uso = uso; // mañana, noche, todo el día
+        this.marca = marca;
+        this.tipo = tipo; // crema, limpiador, etc
+    }
+}
+
+// Array de productos
+
+export const catalogo = [
```

Comment:
Se me ocurre algo así:
```suggestion
export class Catalogo {
    constructor(productos) {
        this.productos = productos;
    }

    buscarPorId(id) {
        for (let i = 0; i < this.productos.length; i++) {
            if (this.productos[i].id === id) {
                return this.productos[i];
            }
        }
        return null;
    }

    obtenerTodos() {
        return this.productos;
    }
}```